components/eamxx/src/physics/mam/eamxx_mam_aci_functions.hpp

```diff
@@ -0,0 +1,726 @@
+                state_q_work_loc(icol, klev, icnst) = state_q_at_lev_col[icnst];
+              }
+
+              // get qqcw at a grid cell (col,lev)
+              // NOTE: The layout for qqcw array is based on mam_idx in
+              // dropmixnuc. To mimic that, we are using the following for-loops
+              int ind_qqcw = 0;
+              for(int m = 0; m < mam_coupling::num_aero_modes(); ++m) {
+                qqcw_view[ind_qqcw](klev) =
+                    dry_aero.cld_aero_nmr[m](icol, klev);
+                ++ind_qqcw;
+                for(int a = 0; a < mam_coupling::num_aero_species(); ++a) {
+                  if(dry_aero.cld_aero_mmr[m][a].data()) {
+                    qqcw_view[ind_qqcw](klev) =
+                        dry_aero.cld_aero_mmr[m][a](icol, klev);
+                    ++ind_qqcw;
+                  }
+                }
+              }
+            });
+
+        mam4::ndrop::dropmixnuc(
+            team, dt, ekat::subview(T_mid, icol), ekat::subview(p_mid, icol),
+            ekat::subview(p_int, icol), ekat::subview(pdel, icol),
+            ekat::subview(rpdel, icol),
+            // in zm[kk] - zm[kk+1], for pver zm[kk-1] - zm[kk]
+            ekat::subview(zm, icol), ekat::subview(state_q_work_loc, icol),
+            ekat::subview(nc, icol), ekat::subview(kvh, icol),  // kvh[kk+1]
+            ekat::subview(cloud_frac, icol), lspectype_amode, specdens_amode,
+            spechygro, lmassptr_amode, num2vol_ratio_min_nmodes,
+            num2vol_ratio_max_nmodes, numptr_amode, nspec_amode, exp45logsig,
+            alogsig, aten, mam_idx, mam_cnst_idx,
+            ekat::subview(qcld, icol),             // out
+            ekat::subview(wsub, icol),             // in
+            ekat::subview(cloud_frac_prev, icol),  // in
+            qqcw_view,                             // inout
+            ptend_q_view, ekat::subview(tendnd, icol),
+            ekat::subview(factnum, icol), ekat::subview(ndropcol, icol),
+            ekat::subview(ndropmix, icol), ekat::subview(nsource, icol),
+            ekat::subview(wtke, icol), ekat::subview(ccn, icol), coltend_view,
+            coltend_cw_view,
+            // work arrays
+            raercol_cw_view, raercol_view, ekat::subview(nact, icol),
+            ekat::subview(mact, icol), ekat::subview(eddy_diff, icol),
+            ekat::subview(zn, icol), ekat::subview(csbot, icol),
+            ekat::subview(zs, icol), ekat::subview(overlapp, icol),
+            ekat::subview(overlapm, icol), ekat::subview(eddy_diff_kp, icol),
+            ekat::subview(eddy_diff_km, icol), ekat::subview(qncld, icol),
+            ekat::subview(srcn, icol), ekat::subview(source, icol),
+            ekat::subview(dz, icol), ekat::subview(csbot_cscen, icol),
+            ekat::subview(raertend, icol), ekat::subview(qqcwtend, icol));
+      });
+}
+
+// Update cloud borne aerosols
+void update_cloud_borne_aerosols(
+    haero::ThreadTeamPolicy team_policy,
```

Comment:
Why passing a policy, if all work is done serially on host?

Reply:
Thanks. Done